### PR TITLE
New MythicMob Tags

### PR DESCRIPTION
**Pull request by heypr**
# Additions/changes

- MythicMobsBridgeTags now uses new tag format.
- New `<mythicmobs.item_ids>` tag.
- New `<mythicmobs.skills>` tag.
- New `<mythicmobs.mob_ids>` tag.
- New `<mythicmobs.damage_modifiers[<mob_id>]>` tag.
- New `<mythicmobs.mob_path[<mob_id>]>` tag.
- New `<mythicmobs.item_path[<item_id>]>` tag.
- New `<mythicmobs.packs>` tag.

# Dependency

- MythicMob dependency bumped, ``5.2.1`` --> ``5.3.5``.

# Author
- [Changes originally authored by 0TickPulse](https://github.com/DenizenScript/Depenizen/pull/392)

### CODE REVIEW

File: src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java [hunk 1]
```diff
@@ -1,48 +1,172 @@
...
+            // @plugin Depenizen, MythicMobs
+            // @description
+            // Returns a map of the damage modifiers for a MythicMob from a Mob ID.
+            // -->
+            tagProcessor.registerTag(MapTag.class, ElementTag.class, "damage_modifiers", (attribute, object, id) -> {
+                MythicMob mob = getMythicMob(id.asString());
+                if (mob == null) {
+                    return null;
+                }
+                return new MapTag(mob.getDamageModifiers().entrySet().stream().map((e) -> Map.entry(new StringHolder(e.getKey()), new ElementTag(e.getValue()))).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
```
Comment (tal5) [inline src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java:129]: This should probably be a proper loop

File: src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java [hunk 2]
```diff
@@ -1,48 +1,172 @@
...
+            });
+
+            // <--[tag]
+            // @attribute <mythicmobs.packs>
+            // @returns ListTag
+            // @plugin Depenizen, MythicMobs
+            // @description
+            // Returns a list of all Mythic pack IDs.
+            // -->
+            tagProcessor.registerTag(ListTag.class, "packs", (attribute, object) -> new ListTag(MythicBukkit.inst().getPackManager().getPacks(), (pack) -> new ElementTag(pack.getName())));
```
Comment (tal5) [inline src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java:166]: Should still be on a new line (`-> {`).
Also, should make these `ElementTag`s plain-text

File: src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java [hunk 3]
```diff
@@ -51,45 +175,12 @@ public void init() {
         PropertyParser.registerProperty(MythicMobsPlayerProperties.class, PlayerTag.class);
         ScriptEvent.registerScriptEvent(MythicMobsDeathEvent.class);
         ScriptEvent.registerScriptEvent(MythicMobsSpawnEvent.class);
-        ScriptEvent.registerScriptEvent(MythicMobsDespawnEvent.class);
```
Comment (tal5) [inline src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java:54]: This doesn't seem like an intentional removal?

File: src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java [hunk 4]
```diff
@@ -51,45 +175,12 @@ public void init() {
...
-        // <--[data]
-        // @name not_switches
-        // @values mythic_mob, mythic_item
-        // -->
-        ScriptEvent.ScriptPath.notSwitches.add("mythic_mob");
-        EntityTag.specialEntityMatchables.add("mythic_mob");
-        BukkitScriptEvent.entityCouldMatchPrefixes.add("mythic_mob");
-        ScriptEvent.ScriptPath.notSwitches.add("mythic_item");
-        BukkitScriptEvent.itemCouldMatchableText.add("mythic_item");
-        BukkitScriptEvent.itemCouldMatchPrefixes.add("mythic_item");
```
Comment (tal5) [inline src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java:92]: These shouldn't be removed either

File: src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java [hunk 5]
```diff
@@ -90,7 +220,6 @@ public void init() {
         ScriptEvent.ScriptPath.notSwitches.add("mythic_item");
         BukkitScriptEvent.itemCouldMatchableText.add("mythic_item");
         BukkitScriptEvent.itemCouldMatchPrefixes.add("mythic_item");
-
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java:93]: this line should not have been removed

File: src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java [hunk 6]
```diff
@@ -105,7 +234,7 @@ public void init() {
             }
             String name = attribute.getParam();
             Optional<MythicItem> itemOpt = MythicBukkit.inst().getItemManager().getItem(name);
-            if (!itemOpt.isPresent()) {
+            if (itemOpt.isEmpty()) {
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java:237]: what

File: src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java [hunk 7]
```diff
@@ -37,12 +40,139 @@
...
+                MythicMob mob = getMythicMob(id.asString());
+                if (mob == null) {
+                    return null;
+                }
+                MapTag damageModifiers = new MapTag();
+                for (Map.Entry<String, Double> entry : mob.getDamageModifiers().entrySet()) {
+                    damageModifiers.putObject(entry.getKey(), new ElementTag(entry.getValue()));
+                }
+                return damageModifiers;
+        });
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java:134]: spaced weird

File: src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java [hunk 8]
```diff
@@ -37,12 +40,139 @@
...
+            // @attribute <mythicmobs.packs>
+            // @returns ListTag
+            // @plugin Depenizen, MythicMobs
+            // @description
+            // Returns a list of all Mythic pack IDs.
+            // -->
+            tagProcessor.registerTag(ListTag.class, "packs", (attribute, object) -> {
+                return new ListTag(MythicBukkit.inst().getPackManager().getPacks(), (pack) -> new ElementTag(pack.getName(), true));
+            });
+        };
```
Comment (mcmonkey4eva) [inline src/main/java/com/denizenscript/depenizen/bukkit/bridges/MythicMobsBridge.java:173]: semicolon shouldn't be here

File: pom.xml [hunk 9]
```diff
@@ -222,7 +222,7 @@
         <dependency>
             <groupId>io.lumine.xikage</groupId>
             <artifactId>mythicmobs</artifactId>
-            <version>5.2.1</version>
+            <version>5.3.5</version>
```
Comment (mcmonkey4eva) [inline pom.xml:225]: This version doesn't exist, latest is 5.3.3

Comment (mcmonkey4eva) [reply pom.xml:225]: Oh, no, MythicMobs just has trash file distribution, nevermind

**Comment by mcmonkey4eva** [comment]
This is uncompilable untested nonsense

**Pull request closed without merging**